build-tools/packages/build-cli/src/commands/generate/changeset.ts

```diff
@@ -0,0 +1,279 @@
+			this.error(`No changed packages when compared to ${branch}.`, { exit: 1 });
+		}
+
+		if (changedReleaseGroups.length > 1) {
+			this.warning(
+				`More than one release group changed when compared to ${branch}. Is this expected?`,
+			);
+		}
+
+		const choices: Choice[] = [];
+
+		// Handle changed release groups first so they show up in the list first.
+		for (const rgName of changedReleaseGroups) {
+			const rg = context.repo.releaseGroups.get(rgName);
+			if (rg === undefined) {
+				this.error(`Release group ${rgName} not found in repo config`, { exit: 1 });
+			}
+
+			choices.push(
+				{ title: `${chalk.bold(rg.kind)}`, heading: true, disabled: true },
+				...rg.packages
+					.filter((pkg) => (all ? true : isIncludedByDefault(pkg)))
+					.sort((a, b) =>
+						a.nameUnscoped < b.nameUnscoped ? -1 : a.name === b.name ? 0 : 1,
+					)
+					.map((pkg) => {
+						const changed = changedPackages.some((cp) => cp.name === pkg.name);
+						return {
+							title: changed
+								? `${pkg.name} ${chalk.red.bold("(changed)")}`
+								: pkg.name,
+							value: pkg,
+							selected: changed,
+						};
+					}),
+			);
+		}
+
+		// Next list independent packages in a group
+		choices.push({ title: chalk.bold("Independent Packages"), heading: true, disabled: true });
+		for (const pkg of context.independentPackages) {
+			if (!all && !isIncludedByDefault(pkg)) {
+				continue;
+			}
+			const changed = changedPackages.some((cp) => cp.name === pkg.name);
+			choices.push({
+				title: changed ? `${pkg.nameColored} ${chalk.red.bold("(changed)")}` : pkg.name,
+				value: pkg,
+				selected: changed,
+			});
+		}
+
+		// Finally list the remaining (unchanged) release groups and their packages
+		for (const rg of context.repo.releaseGroups.values()) {
+			if (!changedReleaseGroups.includes(rg.kind)) {
+				choices.push(
+					{ title: `${chalk.bold(rg.kind)}`, heading: true, disabled: true },
+					...rg.packages
+						.filter((pkg) => (all ? true : isIncludedByDefault(pkg)))
+						.sort((a, b) =>
+							a.nameUnscoped < b.nameUnscoped ? -1 : a.name === b.name ? 0 : 1,
+						)
+						.map((pkg) => {
+							return {
+								title: pkg.name,
+								value: pkg,
+								selected: false,
+							};
+						}),
+				);
+			}
+		}
+
+		const response = await prompts({
+			choices: [...choices, { title: " ", heading: true, disabled: true }],
+			instructions: INSTRUCTIONS,
+			message: "Choose which packages to include in the changeset. Type to filter the list.",
+			name: "selectedPackages",
+			optionsPerPage: 10,
+			type: uiMode === "default" ? "autocompleteMultiselect" : "multiselect",
+			onState: (state: any) => {
+				// eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+				if (state.aborted) {
+					process.nextTick(() => this.exit(1));
+				}
+			},
+		});
+
+		const selectedPackages: Package[] = response.selectedPackages;
+		const bumpType = getDefaultBumpTypeForBranch(branch) ?? "minor";
```

Comment:
Do we have any interest in throwing a warning message for changesets introduced for minor bumps? Something to warn users that we shouldn't introduce breaking changes outside of major bumps

Reply:
Changesets should be present for any notable change, not just breaking changes. Deprecations, for example.